### ISSUE TITLE
Environment variables not correctly set during optimisation runs

### ISSUE THREAD

**Issue by casimp** [guildai/guildai#187]
Environment variables differ between standard runs and optimisation runs. For instance, with the following setup:

```
- model: label_bug
  description: MWE for env variable bug
  operations:
    train:
      main: env_print
      flags-dest: globals
      flags:
        batch_size: 16
```

```
# env_print.py
import os

print('PROJECT_DIR:', os.getenv('PROJECT_DIR'))
print('GUILD_OP:', os.getenv('GUILD_OP'))
```

An individual run (`guild run train batch_size=1`) outputs:

```
PROJECT_DIR: /home/user/guild_test
GUILD_OP: label_bug:train
```

whereas `guild run train batch_size=[1,]` outputs:

```
PROJECT_DIR: 
GUILD_OP: +

```

**Comment by gar1t** [comment]
Thanks for the report! I create a resolution doc for this here:

https://github.com/guildai/issue-resolution/tree/master/187-environment-variables-not-correctly-set-during-optimisation-runs

This is fixed in master and will be available in the next prerelease.

**Comment by gar1t** [comment]
The fix is now generally available in 0.7.0. If this is still an issue, please re-open and we'll sort it out. Thanks again for the report!
